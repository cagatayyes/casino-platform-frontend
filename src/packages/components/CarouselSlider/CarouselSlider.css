.carouselSlider.swiper {
    padding-bottom: 24px;
}

.carouselSlider .swiper-slide {
    width: 596px !important;
    border-radius: 10px;
}

.carouselSlider .swiper-pagination-bullet {
    width: 10px;
    height: 3px;
    background-color: #303750;
    border-radius: 3px;
    opacity: 1;
}

.carouselSlider .swiper-pagination-bullet-active {
    background-color: #edf1ff;
    width: 17px;
}

.carouselSlider .swiper-pagination {
    bottom: -24px !important;
    margin-bottom: 20px;
}

.carouselSlider.navigation-next,
.carouselSlider.navigation-prev {
    display: flex;
    align-items: center;
    justify-content: center;
    position: absolute;
    z-index: 9;
    top: 45%;
    transform: translateY(-45%);
    width: 22px;
    height: 16px;
    color: #000000;
    background-color: #ffffff;
    border: 1px solid #fff;
    border-radius: 3px;
}

.carouselSlider.navigation-prev {
    left: 8px;
}

.carouselSlider.navigation-next {
    right: 8px;
}

.carouselSlider.swiper-button-disabled {
    display: none;
}

@media screen and (max-width: 768px) {
    .carouselSlider.navigation-next,
    .carouselSlider.navigation-prev {
        display: none;
    }

    .carouselSlider .swiper-slide:first-of-type {
        padding-left: 5px;
    }

    .carouselSlider .swiper-slide:last-of-type {
        padding-right: 5px;
    }

    .carouselSlider .swiper-slide {
        width: 350px !important;
        transform: scaleY(0.94) !important;
    }

    .carouselSlider .swiper-slide-active {
        transform: scaleY(1) !important;
    }
}
