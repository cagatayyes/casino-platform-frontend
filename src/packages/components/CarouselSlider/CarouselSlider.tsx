import React, { ReactElement } from 'react';
import { Swiper, SwiperClass, SwiperSlide } from 'swiper/react';
import { Pagination, Navigation } from 'swiper/modules';
import { useScreenDetector } from 'casino/hooks/useScreenDetector';
import 'swiper/css';
import 'swiper/css/pagination';
import 'swiper/css/navigation';
import './CarouselSlider.css';
import { useDispatchCustomEvent } from 'casino/hooks/useDispatchEvents';
import { EventActions } from 'casino/constants';

// eslint-disable-next-line @typescript-eslint/naming-convention
export interface ICarouselSlider {
	children: ReactElement[];
	maxheight?: number | string;
	maxwidth?: number | string;
	sliderheight?: number | string;
	sliderwidth?: number | string;
	slides?: unknown[];
}

export interface ICarouselSliderItemProps extends React.ComponentProps<'div'> {
	className?: string;
}

// eslint-disable-next-line no-unused-vars
export const CarouselSlider: React.MemoExoticComponent<(props: ICarouselSlider) => React.JSX.Element> = React.memo((props: ICarouselSlider) => {
	const { children, maxwidth = 'auto', slides } = props;
	const { isMobile } = useScreenDetector();
	const { dispatchCustomEvent } = useDispatchCustomEvent();

	return (
		<div className='relative mt-[10px]'
			style={{
				maxWidth: maxwidth,
			}}
			data-testid='homePageBannerCarousel'>
			<Swiper
				modules={[Pagination, Navigation]}
				data-testid='lobby-carousel-slider'
				navigation={{
					nextEl: '.navigation-next',
					prevEl: '.navigation-prev',
					disabledClass: 'swiper-button-disabled'
				}}
				breakpoints={{
					768: {
						slidesPerView: 'auto',
						slidesPerGroup: 1,
						centeredSlides: true,
						centerInsufficientSlides: true,
						centeredSlidesBounds: true,
						spaceBetween: 7,
					},
				}}
				slidesPerView={isMobile ? 'auto' : 2}
				slidesPerGroup={1}
				centeredSlides={true}
				centerInsufficientSlides={true}
				centeredSlidesBounds={true}
				autoplay={{
					delay: 3000,
					pauseOnMouseEnter: true
				}}
				pagination={
					{ clickable: true }
				}
				spaceBetween={7}
				className="carouselSlider bg-[var(--main-bg-color)]"
				{...props}
				style={{
					maxHeight: 'auto',
					maxWidth: maxwidth
				}}
				onClick={(swiper: SwiperClass) => {
					const clickedIndex = swiper.clickedIndex || 0;
					const clickedSlide = slides?.find((slide: unknown, index: number) => index === clickedIndex) as { name: string };

					if (clickedSlide) {
						const data = {
							order: clickedIndex + 1,
							name: clickedSlide.name,
						};

						dispatchCustomEvent(EventActions.sliderClick, data);
					}
				}}
			>
				{React.Children.map(children, (child: React.JSX.Element, index: number) => (
					<SwiperSlide
						data-testid='lobby-carousel-slider-item'
						key={index}
						{...child.props}
					>
						{child}
					</SwiperSlide>
				))}
			</Swiper>

			<button className="carouselSlider navigation-prev arrow">
				<span className='text-[18px] material-symbols-outlined'>
					keyboard_arrow_left
				</span>
			</button>

			<button className="carouselSlider navigation-next arrow">
				<span className='text-[18px] ml-[auto] material-symbols-outlined'>
					keyboard_arrow_right
				</span>
			</button>
		</div >
	);
});


// eslint-disable-next-line no-unused-vars
export const CarouselSliderItem: React.MemoExoticComponent<(props: ICarouselSliderItemProps) => React.JSX.Element> =
	React.memo(({ children }: ICarouselSliderItemProps) => (<>{children}</>));

CarouselSlider.displayName = 'CarouselSlider';
CarouselSliderItem.displayName = 'CarouselSliderItem';

export default CarouselSlider;
