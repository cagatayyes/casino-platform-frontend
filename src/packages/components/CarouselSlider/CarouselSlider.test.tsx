/* eslint-disable react/prop-types */
jest.mock('swiper/react', () => ({
	Swiper: ({ children }): JSX.Element => <div>{children}</div>,
	SwiperSlide: ({ children }): JSX.Element => <div>{children}</div>
}));

jest.mock('swiper/modules', () => ({
	Swiper: ({ children }): JSX.Element => <div>{children}</div>,
	SwiperSlide: ({ children }): JSX.Element => <div>{children}</div>
}));

jest.mock('casino/hooks/useScreenDetector', () => ({
	// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
	useScreenDetector: () => ({ isMobile: false })
}));

jest.mock('casino/hooks/useDispatchEvents', () => ({
	useDispatchCustomEvent: (): { dispatchCustomEvent: jest.Mock } => ({ dispatchCustomEvent: jest.fn() })
}));

/* eslint-disable react/prop-types */
import React from 'react';
import { render, screen } from '@testing-library/react';
import CarouselSlider from './CarouselSlider';

describe('CarouselSlider', () => {
	it('should render successfully', () => {
		const component = render(
			<CarouselSlider>
				<div>Slide 1</div>
				<div>Slide 2</div>
				<div>Slide 3</div>
				<div>Slide 4</div>
				<div>Slide 5</div>
				<div>Slide 6</div>
			</CarouselSlider>,
		);
		expect(screen.getByTestId('homePageBannerCarousel')).toBeInTheDocument();
		expect(component).toMatchSnapshot();
	});

	it('should render child elements', () => {
		const component = render(
			<CarouselSlider>
				<div>Slide 1</div>
				<div>Slide 2</div>
				<div>Slide 3</div>
				<div>Slide 4</div>
				<div>Slide 5</div>
				<div>Slide 6</div>
			</CarouselSlider>,
		);
		expect(screen.getByText('Slide 1')).toBeInTheDocument();
		expect(screen.getByText('Slide 2')).toBeInTheDocument();
		expect(component).toMatchSnapshot();
	});
});
