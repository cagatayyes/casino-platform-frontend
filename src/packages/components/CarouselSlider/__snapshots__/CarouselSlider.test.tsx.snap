// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselSlider should render child elements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative mt-[10px]"
        data-testid="homePageBannerCarousel"
        style="max-width: auto;"
      >
        <div>
          <div>
            <div>
              Slide 1
            </div>
          </div>
          <div>
            <div>
              Slide 2
            </div>
          </div>
          <div>
            <div>
              Slide 3
            </div>
          </div>
          <div>
            <div>
              Slide 4
            </div>
          </div>
          <div>
            <div>
              Slide 5
            </div>
          </div>
          <div>
            <div>
              Slide 6
            </div>
          </div>
        </div>
        <button
          class="carouselSlider navigation-prev arrow"
        >
          <span
            class="text-[18px] material-symbols-outlined"
          >
            keyboard_arrow_left
          </span>
        </button>
        <button
          class="carouselSlider navigation-next arrow"
        >
          <span
            class="text-[18px] ml-[auto] material-symbols-outlined"
          >
            keyboard_arrow_right
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative mt-[10px]"
      data-testid="homePageBannerCarousel"
      style="max-width: auto;"
    >
      <div>
        <div>
          <div>
            Slide 1
          </div>
        </div>
        <div>
          <div>
            Slide 2
          </div>
        </div>
        <div>
          <div>
            Slide 3
          </div>
        </div>
        <div>
          <div>
            Slide 4
          </div>
        </div>
        <div>
          <div>
            Slide 5
          </div>
        </div>
        <div>
          <div>
            Slide 6
          </div>
        </div>
      </div>
      <button
        class="carouselSlider navigation-prev arrow"
      >
        <span
          class="text-[18px] material-symbols-outlined"
        >
          keyboard_arrow_left
        </span>
      </button>
      <button
        class="carouselSlider navigation-next arrow"
      >
        <span
          class="text-[18px] ml-[auto] material-symbols-outlined"
        >
          keyboard_arrow_right
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CarouselSlider should render successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="relative mt-[10px]"
        data-testid="homePageBannerCarousel"
        style="max-width: auto;"
      >
        <div>
          <div>
            <div>
              Slide 1
            </div>
          </div>
          <div>
            <div>
              Slide 2
            </div>
          </div>
          <div>
            <div>
              Slide 3
            </div>
          </div>
          <div>
            <div>
              Slide 4
            </div>
          </div>
          <div>
            <div>
              Slide 5
            </div>
          </div>
          <div>
            <div>
              Slide 6
            </div>
          </div>
        </div>
        <button
          class="carouselSlider navigation-prev arrow"
        >
          <span
            class="text-[18px] material-symbols-outlined"
          >
            keyboard_arrow_left
          </span>
        </button>
        <button
          class="carouselSlider navigation-next arrow"
        >
          <span
            class="text-[18px] ml-[auto] material-symbols-outlined"
          >
            keyboard_arrow_right
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="relative mt-[10px]"
      data-testid="homePageBannerCarousel"
      style="max-width: auto;"
    >
      <div>
        <div>
          <div>
            Slide 1
          </div>
        </div>
        <div>
          <div>
            Slide 2
          </div>
        </div>
        <div>
          <div>
            Slide 3
          </div>
        </div>
        <div>
          <div>
            Slide 4
          </div>
        </div>
        <div>
          <div>
            Slide 5
          </div>
        </div>
        <div>
          <div>
            Slide 6
          </div>
        </div>
      </div>
      <button
        class="carouselSlider navigation-prev arrow"
      >
        <span
          class="text-[18px] material-symbols-outlined"
        >
          keyboard_arrow_left
        </span>
      </button>
      <button
        class="carouselSlider navigation-next arrow"
      >
        <span
          class="text-[18px] ml-[auto] material-symbols-outlined"
        >
          keyboard_arrow_right
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
